src/main/java/appeng/core/worlddata/SpatialDimensionManager.java

```diff
@@ -0,0 +1,222 @@
+		if( cell != null )
+		{
+			return cell.owner;
+		}
+		return -1;
+	}
+
+	@Override
+	public BlockPos getCellDimensionOrigin( int cellStorageId )
+	{
+		if( this.isCellDimension( cellStorageId ) )
+		{
+			return getBlockPosFromId( cellStorageId );
+		}
+		return null;
+	}
+
+	@Override
+	public BlockPos getCellContentSize( int cellStorageId )
+	{
+		StorageCellData cell = this.spatialData.get( cellStorageId );
+		if( cell != null )
+		{
+			return cell.contentDimension;
+		}
+		return null;
+	}
+
+	@Override
+	public boolean hasCapability( Capability<?> capability, EnumFacing facing )
+	{
+		return capability == Capabilities.SPATIAL_DIMENSION;
+	}
+
+	@Override
+	public <T> T getCapability( Capability<T> capability, EnumFacing facing )
+	{
+		if( capability == Capabilities.SPATIAL_DIMENSION )
+		{
+			return (T) this;
+		}
+		return null;
+	}
+
+	@Override
+	public NBTTagCompound serializeNBT()
+	{
+		final NBTTagCompound ret = new NBTTagCompound();
+		final NBTTagList list = new NBTTagList();
+
+		for( Map.Entry<Integer, StorageCellData> entry : this.spatialData.entrySet() )
+		{
+			final NBTTagCompound nbt = entry.getValue().serializeNBT();
+			nbt.setInteger( "id", entry.getKey() );
+			list.appendTag( nbt );
+		}
+		ret.setTag( "spatial_data", list );
+		return ret;
+	}
+
+	@Override
+	public void deserializeNBT( NBTTagCompound nbt )
+	{
+		if( nbt.hasKey( "spatial_data" ) )
+		{
+			final NBTTagList list = (NBTTagList) nbt.getTag( "spatial_data" );
+
+			this.spatialData.clear();
+			for( int i = 0; i < list.tagCount(); ++i )
+			{
+				final NBTTagCompound entry = list.getCompoundTagAt( i );
+				final StorageCellData data = new StorageCellData();
+				final int id = entry.getInteger( "id" );
+				data.deserializeNBT( entry );
+				this.spatialData.put( id, data );
+			}
+		}
+	}
+
+	private int getNextId()
+	{
+		return this.spatialData.keySet().stream().max( Integer::compare ).orElse( -1 ) + 1;
+	}
+
+	private BlockPos getBlockPosFromId( int id )
+	{
+		int signBits = id & 0b11;
+		int offsetBits = id >> 2;
+		int offsetScale = 1;
+		int posx = MAX_CELL_DIMENSION / 2;
+		int posz = MAX_CELL_DIMENSION / 2;
+
+		// find quadrant
+		while( offsetBits != 0 )
+		{
+			posx += MAX_CELL_DIMENSION * offsetScale * ( offsetBits & 0b01 );
+			posz += MAX_CELL_DIMENSION * offsetScale * ( offsetBits >> 1 & 0b01 );
+
+			offsetBits >>= 2;
+			offsetScale <<= 1;
+		}
+
+		// mirror in one of 4 directions
+		posx *= ( signBits & 0x01 ) - 1;
+		posz *= ( signBits >> 1 & 0x01 ) - 1;
+
+		// offset from cell center
+		posx -= 64;
+		posz -= 64;
+
+		return new BlockPos( posx, 64, posz );
+	}
+
+	private void initCellArea( StorageCellData cell )
+	{
+		// TODO lets build a wall
+	}
+
+	private void clearCellArea( StorageCellData cell )
+	{
+		// TODO clear all blocks in area?
+	}
+
+	private static class StorageCellData implements INBTSerializable<NBTTagCompound>
+	{
+		public BlockPos contentDimension;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make contentDimension a static final constant or non-public and provide accessors if needed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)